src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lateralSide;
+    private double upperBase;
+    private double bottomBase;
+    private double height;
+    private double triangleLeg;
+
+    public IsoscelesTrapezoid() {
+    }
+
+    public IsoscelesTrapezoid(double upperBase, double bottomBase, double lateralSide) {
+        this.typeOfFigure = TypeOfFigure.ISOSCELES_TRAPEZOID;
+        this.area = calculateArea(upperBase, bottomBase, triangleLeg, height);
```

Comment:
The method `calculateArea` should be a class-level method, not an instance method. Also, consider whether you need to pass all these parameters to `calculateArea` since you can access class fields directly.